### ISSUE TITLE
Understanding for Contrast does not call out Color Blindness

### ISSUE THREAD

**Issue by bruce-usab** [w3c/wcag#2033]
The [Understanding document](https://www.w3.org/WAI/WCAG22/Understanding/contrast-minimum.html) for minimum Contrast (SC 1.4.3) does not explicitly mention that the 4.5:1 metric well addresses the needs of people with (the four types of) red-green color blindness and (the two types) of blue-yellow color blindness.

I believe this to be a problem, as the normative requirement lacks obvious face validity.  My own personal experience is that I have had to explain this feature dozens of times.  I think is would be better for our authoritative reference to include this detail.

I will submit a pull request with a minimal edit.

**Comment by bruce-usab** [comment]
_Originally posted by @GreggVan in https://github.com/w3c/wcag/issues/695#issuecomment-916568064_

> But remember that in the end you need 
> 1. to account for color blindness as well as low vision 
> 2. account for 3 layer contrast  (A contrasts with B that contrasts with A and C) 
> 3. that you need enough colors left after you do 1 and 2 to be acceptable to designers.

For (3) is it worth mentioning that of the 2.8x10^14 RGB pairs, _X_ provide contrast of 4.5:1 or higher?

**Comment by Myndex** [comment]
Hi Bruce @bruce-usab and @mbgower 

I also responded BRIEFLY in the pull request with specific change requests...  #2034 

### The Ableist Term "Color Blind"
Commenting here, includes some of what I mention in the pull request.

Before I get into "technical stuff" I want to being on an agenda item I want to discuss for on the LVTF, namely the term "color blind" ... I am hoping we can start normalizing something that is not so ableist, as 99.98% of CVD types are **not color blind,** they are color limited. But they still see colors.

_AS SUCH_, I object to adding that line in the PR as is, can instead it say something like

_"sometimes, though incorrectly, referred to as color blindness"_ 

or perhaps (my preference but longer) 

_**"formerly referred to as 'color blind', though in fact the vast majority of color-limited vision types do differentiate at least some colors"**_

## Some Additional Technical Comments:

For the first line (18) in the PR, can we add "_...so long as luminance contrast is high enough to support rapid reading._" the suggested edit of the first line then reads:

**For people without color deficiencies, hue and saturation have minimal or no effect on legibility so long as luminance contrast is high enough to support rapid reading** _(Knoblauch et al., 1991)._

**Because:** the issue is a bit more complicated, and there are definitely color combinations that can impact readability, particularly when there is insufficient luminance contrast. The line I suggest above helps to clarify and is echoed also by Legge in his Psychophysics of reading, IX.

### Second Line (19)
It reads "Color deficiencies can affect luminance contrast somewhat." 

Can we change to: 
**_"Color deficiencies can affect luminance contrast in some cases, as can other impairment types."_** 

Actually nearly all forms of CVD have otherwise normal vision and normal contrast, the notable exception being BCM (which may soon be removed from the CVD classification). CVD do suffer in contrast for the color area they are deficient, such as protan having poor contrast in reds.

Deutans (by far the most common) have an essentially normal CS as they have L and S cones that cover the same spectral area as normal vision. Their deficit is in M cone which limits their differentiation of colors that require the M cone or the M+L opponent process.

And also: many impairments affect contrast, IMO should be mentioned as noted.

### CVD
I don't want to get _too_ deep into whether or not any CVD types are functionally assisted with 1.4.3 or 1.4.6. In a classical design sense, body text should be about 10:1, regardless of CVD or not. The 4.5:1 ratio allows pure red `#F00` against black `#000` to pass, but this is actually a fail for someone with protanopia as I discuss in this article:

#### [_The Lighter Side of Dark Backgrounds_](https://gist.github.com/Myndex/c30dba273aa5eca426ad9f5200917c9d)

The main upshot is that adequate luminance contrast is needed by ALL sighted readers, and the 2.x contrast method does not do anything "special" to specifically assist any particular CVD type, as I discuss in the above article with demonstrations.


### BCM
And Blue Cone Monochromacy is not assisted in any real way with 1.4.3 or 1.4.6. BCM needs assistive technologies as they are co-morbid with low vision, not to mention photophobia. For visual readability, need at a minimum to modify or adjust their screen. Moreover, a page design that is using blue to help add discrimination for a protan or deutan, will negatively affect what BCM sees, outside of a custom CSS style sheet.

I suggest removing any claims of assisting Monochromacy because that is unsupportable.


### Middle Grey & Colors
> _(3) is it worth it to add mention that of the 2.8x10^14 RGB pairs, X provide contrast of 4.5:1 or higher?_

I'm on record separately, and with supporting empirical evidence that I made publicly available, that a large proportion of the color pairs that "succeed" under 1.4.3 should instead fail.

I would like to suggest that selling a point that is not actually accurate should be avoided.


### On the subject of "three way":
> _account for 3 layer contrast (A contrasts with B that contrasts with A and C)_

Unfortunately, the way that manifests in 1.4.3 is neither helpful for readability nor design. Yes, multi-level colors is needed in design, and can be accommodated on an sRGB display. The "three way" using 4.5:1 promotes a middle grey that is too dark, and not actual middle grey, and certainly not middle contrast, on an sRGB display under typical viewing conditions.

The high spatial frequency of body text is not well served by 4.5:1. Meanwhile, a large block of color on a ` <div> ` or ` <button> `does not usually need anything as high as 4.5:1 (the TEXT does, but the shape does not).

The point is that "three way" in a manner that is useful for design is best achieved by looking at the contrast needs for the specific element.


**_MOREOVER_**
The colors that 1.4.3 contrast most often does a "false pass" are the darker pairs, again as illustrated in the above linked article. When 2.2 was moving forward, I had been working on some possible SCs to address some of this, though was too swamped getting APCA ready for Silver. One of those unfinished SCs was:

**"One of the two colors of a color pair must be lighter than #A0A0A0"** 

As a interim measure to help fix some of the more serious problems with poor contrast on some websites that still pass WCAG 2.x.


## TL;DR

- Can we avoid using "color blind" as part of a push to re-label is as "color limited vision" ?
- Can we change the first line to indicate that best reading needs high enough luminance contrast?
- I generally maintain my objections (since April 2019) regarding the efficacy of WCAG 2.x contrast. 
- I'm not quite sure what's going on here with the out of the blue comment in #695 regarding two year old posts that were nevertheless privately discussed in 2019 yet are not aware of the last two years of developments. I'll save that for Thursday.

Thank you!

Andy


_Andrew Somers_      
_AGWG Invited Expert_      
_Myndex Color Science Researcher_

**Comment by Myndex** [comment]
## REGARDING COLOR CONTRAST

I wanted to review some of the literature before commenting further on the line:

> _"...For people without color deficiencies, hue and saturation have minimal or no effect on legibility as assessed by reading performance (Knoblauch et al., 1991)."_

While I suggested a minor edit in the previous post, I wanted to comment further as that line seemed a bit askew and out of context relative to some more recent research.

Without doing a super-deep dive, but citing Legge et al from **Psychophysics of Reading XI:**

- Subjects with low vision do better with luminance contrast than with color (hue/sat) contrast.
    - This is using very large letters that subtend 6° on the retina.
    - This is approximately a 280px x-height, aka a 560px font.
- Subjects with normal vision read giant 6° letters as fast for color or luminance contrast.
- BUT Normal vision read 1° letters faster for luminance contrast than for color contrast.
    - 1° is a 45px x-height aka a 90-px font, still very large.
    - scaling the color contrast showed in overlay a similar curve to the luminance contrast, but no scaling was needed for the normal vision and 6° letters.
    - There was some discussion regarding scaling color contrast based on value over threshold, though this was not needed for the giant letters.
- Color and luminance contrast are NOT additive, so having both did not improve reading speed.
- The color contrast tested was limited to red/green. Other colors (such as red/blue which are known to be a problem) were not tested.

### There are important implications, which I discuss:

 The visual angle of a typical  16px font is about ⅓rd of a degree, or 1/6th of a degree in x-height. 

The normal vision group were tested with letter 6 times normal body text, and 36 times normal body text. The low vision group, only the 36 times letters.

### KEY TAKEAWAYS:
1) The Low Vision group did better with luminance contrast than color contrast.
2) The normal vision group did better with smaller letters and luminance contrast than color contrast, BUT normal size text was not tested, and there are important implications here.

## Spatial Frequency
We know that at higher spatial frequencies, that only luminance contrast is functional. This is because color (hue and chroma) roll off at a much lower frequency than does luminance.

The 6° letters are large enough that for normal vision, they are well within the useable CS curve for chroma contrast, thus it is not surprising that normal vision read the 6° letters similarly for both color or luminance contrast. ALSO Normal vision read the 6° letters SLOWER than the 1° letters overall.

IMPORTANTLY those with low vision did better with the luminance contrast than color, and this should be noted as a key takeaway.

The 1° letters for normal vision are still very much larger than than body text, and near the peak of the luminance CS curve. The fact that there was a shift in scale between luminance and color contrast is expected, due to the different spatial frequency responses.

**Other studies,** not to mention my own empirical studies here in the lab, show that at the high spatial frequencies of COLUMNS OF BODY TEXT i.e. 1/6th of a degree letters, that only luminance contrast is helpful for readability, and in fact some color pairs can cause excessive disability glare.

This is expected due to the frequency response of contrast sensitivity of the chromatic and luminance channels in the visual cortex.

**_SIDE NOTES:_** 6° letters are so large, they are likely recognized only as letter by letter objects, and not via the VFWA (Visual Word Form Area) as such they are read slower than 1° letters by normal vision. The range of letter sizes for _best_ readability is about 0.2° to 2° of visual angle. x-height for a 18px font is just shy of 0.2°, which is in part why I frequently **recommended an 18px font for body text.**


## TL;DR
Ultimately, I want to ask to change the line, as the way it reads it is too general and broadly states something regarding color contrast that, out of context related to glyph size, could be taken and used the wrong way.

**_We need to reinforce that:_**

- For low vision, luminance contrast is the important one. 
    - This is true of course for CVD also.
- And luminance contrast is what is important for small text such as body text.

### _Suggested new statement:_
**For people with normal vision and reading very large letters, hue and saturation have minimal effect on legibility. (Knoblauch et al., 1991) However for people with low vision, for people with color limited vision, luminance contrast is the most important for readability (Legge, Psychophysics of Reading XI). This also holds true for normal vision when using small letters such as in body text. (Barten 2003)**

Thank you


Andy

**Comment by bruce-usab** [comment]
I don't mind including why "color blind" is a misnomer.   But the phrase is in common use, and IMHO it is a defect (i.e., barrier to communication) that Understanding does not mention the term.

**Comment by Myndex** [comment]
Hi Bruce @bruce-usab 

> I don't mind including why "color blind" is a misnomer. But the phrase is in common use, and IMHO it is a defect (i.e., barrier to communication) that Understanding does not mention the term.

Okay, that makes sense... maybe this is the first step to changing this...  if we can at least get the idea out there that the term is incorrect and ableist, that's a step in teh right direction.

**Comment by mraccess77** [comment]
I support limiting the use of "color blind" and when it's used explaining why it's often an inaccurate and problematic term.

**Comment by bruce-usab** [comment]
I am taking another run at this.  It is taking longer than I anticipated...

**Comment by bruce-usab** [comment]
PR to close this issue is ready for AGWG survey.  There is quite a lot of useful information in this Issue thread and conversation on #3240 and #3284 &mdash; but my recommendation is that the need for those be captured in a new issue.

**Comment by bruce-usab** [comment]
Current:
> The intent of this Success Criterion is to provide enough contrast between text and its background so that it can be read by people with moderately low vision (who do not use contrast-enhancing assistive technology). For people without color deficiencies, hue and saturation have minimal or no effect on legibility as assessed by reading performance (Knoblauch et al., 1991). Color deficiencies can affect luminance contrast somewhat. Therefore, in the recommendation, the contrast is calculated in such a way that color is not a key factor so that people who have a color vision deficit will also have adequate contrast between the text and the background.

Proposed:
> The intent of this Success Criterion is to provide enough luminance contrast between text and its background, so that it can be read by people with moderately impaired contrast perception, without the use of contrast-enhancing assistive technology.
> 
> For all consumers of visual content, adequate luminance contrast is needed between text and its background for good readability.  Many different visual impairments can substantially impact contrast sensitivity, requiring more luminance contrast, regardless of color (hue).  For people who are not able to distinguish certain shades of color &mdash; often referred to as <q>color blindness</q> &mdash; hue and saturation have minimal or no effect on legibility as assessed by reading performance (<a href="https://doi.org/10.1364/JOSAA.8.000428" title="Effects of chromatic and luminance contrast on reading">Knoblauch et al., 1991</a>).  Further, the inability to distinguish certain shades of color does not negatively affect luminance contrast perception (<a href="https://pubmed.ncbi.nlm.nih.gov/24103453/" title="Achromatic luminance contrast sensitivity in X-linked color-deficient observers: an addition to the debate">Márta Janáky et al., 2013</a>; <a href="https://link.springer.com/article/10.1007/s10792-018-0881-7" title="Contrast sensitivity of patients with congenital color vision deficiency">Cagri Ilhan et al., 2018</a>).  Therefore, in the recommendation, contrast is calculated in such a way that color (hue) is not a key factor.

**Comment by GreggVan** [comment]
This is not quite correctWhen these were created the contest was calculated to account for BOTH low vision and color blindness together in a person.  (We did the calculations)BestGreggLife is short and dictation is faster.  Excuse errors. On Jul 23, 2023, at 5:40 PM, Bruce Bailey ***@***.***> wrote:﻿
Current:

The intent of this Success Criterion is to provide enough contrast between text and its background so that it can be read by people with moderately low vision (who do not use contrast-enhancing assistive technology). For people without color deficiencies, hue and saturation have minimal or no effect on legibility as assessed by reading performance (Knoblauch et al., 1991). Color deficiencies can affect luminance contrast somewhat. Therefore, in the recommendation, the contrast is calculated in such a way that color is not a key factor so that people who have a color vision deficit will also have adequate contrast between the text and the background.

Proposed:

The intent of this Success Criterion is to provide enough luminance contrast between text and its background, so that it can be read by people with moderately impaired contrast perception, without the use of contrast-enhancing assistive technology.
For all consumers of visual content, adequate luminance contrast is needed between text and its background for good readability.  Many different visual impairments can substantially impact contrast sensitivity, requiring more luminance contrast, regardless of color (hue).  For people who are not able to distinguish certain shades of color — often referred to as color blindness — hue and saturation have minimal or no effect on legibility as assessed by reading performance (Knoblauch et al., 1991).  Further, the inability to distinguish certain shades of color does not negatively affect luminance contrast perception (Márta Janáky et al., 2013; Cagri Ilhan et al., 2018).  Therefore, in the recommendation, contrast is calculated in such a way that color (hue) is not a key factor.

**Comment by bruce-usab** [comment]
Current first sentence under intent:
> The intent of this Success Criterion is to provide enough contrast between text and its background so that it can be read by people with moderately low vision (who do not use contrast-enhancing assistive technology).

So how about?
> The intent of this Success Criterion is to provide enough luminance contrast between text and its background, so that it can be read by people with moderately low vision or impaired contrast perception, without the use of contrast-enhancing assistive technology.

**Comment by Myndex** [comment]
Hi @bruce-usab @GreggVan 


Bruce:
> > _The intent of this Success Criterion is to provide enough luminance contrast between text and its background, so that it can be read by people with moderately low vision or impaired contrast perception, without the use of contrast-enhancing assistive technology._

Hi Bruce, I think this is good, and I like it better here without the parenthetical.


Gregg:
> _This is not quite correct. When these were created the contest was calculated to account for BOTH low vision and color blindness together in a person.  (We did the calculations)_ 

Hi Gregg, one of the tasks we examined in the Visual Contrast group was to understand the ideas and motivation behind the original SC from 2007. I discuss some of that [in thread 1705](https://github.com/w3c/wcag/issues/1705#issuecomment-1027058976).

The last ten years of vision science, and particularly impairments, has expanded understanding. With the current understanding:

1) **VA+CS:** It's not ideal to conflate visual acuity and contrast sensitivity, as there is not a consistent relationship.¹ VA and CS are separate, largely independent visual functions.

2) **Color vision deficiency**, especially deutan and protan, have as good or better luminance contrast sensitivity and visual function overall, compared to standard vision²⁻³. One theory behind this as regards to dichromats, is due to a lower neural noise.

As I've written about⁴, protans experience the one readability-related luminance contrast deficit in the edge case of red-based colors when paired against black, as protanopia sees the sRGB red primary as about 55% darker. As a result, saturated reds, oranges, and purples are generally best paired with white and not black (other impairments notwithstanding).

Readability is about luminance contrast, disregarding color (hue/chroma), as high spatial frequencies, i.e. text, are a function of the luminance channel in the HVS. Hue/Chroma processing is at very low spatial resolution.

An expanding understanding here also, is how correcting for one visual impairment may be detrimental to other visual impairments, pointing to personalization as the ideal solution.

Thank you for reading.


_**References:**_
1) [Relationship Between Acuity and Contrast Sensitivity](https://iovs.arvojournals.org/article.aspx?articleid=2770151): Differences Due to Eye Disease (Low Vision|June 2020)
2) [luminance contrast sensitivity in color-deficient observers](https://pubmed.ncbi.nlm.nih.gov/24103453/) Márta Janáky et al., 2013
3) [Contrast sensitivity of patients with color vision deficiency](https://link.springer.com/article/10.1007/s10792-018-0881-7)) Cagri Ilhan et al., 2018
4) [What’s Red & Black & Not Read All Over?](https://tangledweb.xyz/whats-red-black-also-not-read-573b9c0a97ed) Somers, Jan 2022

**Comment by alastc** [comment]
Closing as superseded by #3240

**Comment by bruce-usab** [comment]
@Myndex, please affirm that your recent reply to GreggV is not requesting additional changes to PR #3284 as I understand you are noting need for subsequent updates.

**Comment by Myndex** [comment]
> @Myndex, please affirm that your recent reply to GreggV is not requesting additional changes to PR #3284 as I understand you are noting need for subsequent updates.

Hi @bruce-usab 

Sorry, yes, I agree with your change, my note to GreggV was (intended) to answer what "I thought" he was discussing?. Sorry for confusion...
